x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java

```diff
@@ -165,26 +184,38 @@ protected final List<Page> oneDriverPerPageList(Iterator<List<Page>> source, Sup
     protected final void assertSimple(DriverContext context, int size) {
         List<Page> input = CannedSourceOperator.collectPages(simpleInput(context.blockFactory(), size));
         // Clone the input so that the operator can close it, then, later, we can read it again to build the assertion.
-        List<Page> inputClone = CannedSourceOperator.deepCopyOf(input);
+        List<Page> origInput = BlockTestUtils.deepCopyOf(input, BlockFactory.getNonBreakingInstance());
         BigArrays bigArrays = context.bigArrays().withCircuitBreaking();
-        List<Page> results = drive(simple(bigArrays).get(context), input.iterator());
-        assertSimpleOutput(inputClone, results);
-        results.forEach(Page::releaseBlocks);
+
+        Operator operator = null;
+        try {
+            operator = simple(bigArrays).get(context);
+            List<Page> results = drive(operator, input.iterator(), context);
+            assertSimpleOutput(origInput, results);
+            // results.forEach(Page::releaseBlocks); // TODO: remove
+            assertThat(bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST).getUsed(), equalTo(0L));
+        } catch (CircuitBreakingException cbe) {
+            if (operator == null) {
+                // if we failed to even create the operator, then release the input pages
```

Comment:
should this just wrap the `operator =` stuff?

Reply:
What causes the operators to fail to build? I'd really love if they didn't fail to build. But I suppose that's life.

Reply:
I simplified and clean up all this.